packages/builder/src/components/backend/DataTable/buttons/grid/GridViewCalculationButton.svelte

```diff
@@ -0,0 +1,259 @@
+  }
+
+  const deleteCalc = idx => {
+    calculations = calculations.toSpliced(idx, 1)
+
+    // Remove any grouping if clearing the last calculation
+    if (!calculations.length) {
+      groupBy = []
+    }
+  }
+
+  const save = async () => {
+    let newSchema = {}
+
+    // Add calculations
+    for (let calc of calculations) {
+      if (!calc.type || !calc.field) {
+        continue
+      }
+      const typeOption = calculationTypeOptions.find(x => x.value === calc.type)
+      const name = `${typeOption.label} ${calc.field}`
+      newSchema[name] = {
+        calculationType: calc.type,
+        field: calc.field,
+        visible: true,
+      }
+    }
+
+    // Add other fields
+    for (let field of Object.keys(schema)) {
+      if (schema[field].calculationType) {
+        continue
+      }
+      newSchema[field] = {
+        ...schema[field],
+        visible: groupBy.includes(field),
+      }
+    }
+
+    // Ensure primary display is valid
+    let primaryDisplay = $definition.primaryDisplay
+    if (!primaryDisplay || !newSchema[primaryDisplay]?.visible) {
+      primaryDisplay = groupBy[0]
+    }
+
+    // Save changes
+    await datasource.actions.saveDefinition({
+      ...$definition,
+      primaryDisplay,
+      schema: newSchema,
+    })
+    await rows.actions.refreshData()
+  }
+</script>
+
+<ActionButton icon="WebPage" quiet on:click={open}>
+  Configure calculations{count ? `: ${count}` : ""}
+</ActionButton>
+
+<Modal bind:this={modal}>
+  <ModalContent
+    title="Calculations"
+    confirmText="Save"
+    size="M"
+    onConfirm={save}
+  >
+    {#if calculations.length}
+      <div class="calculations">
+        {#each calculations as calc, idx}
+          <span>{idx === 0 ? "Calculate" : "and"} the</span>
+          <Select
+            options={getTypeOptions(calc, calculations)}
+            bind:value={calc.type}
+            placeholder={false}
+          />
+          <span>of</span>
+          <Select
+            options={getFieldOptions(calc, calculations, schema)}
+            bind:value={calc.field}
+            placeholder="Column"
+          />
+          <Icon
+            hoverable
+            name="Delete"
+            size="S"
+            on:click={() => deleteCalc(idx)}
+            color="var(--spectrum-global-color-gray-700)"
+          />
+        {/each}
+        <span>Group by</span>
+        <div class="group-by">
+          <Multiselect
+            options={groupByOptions}
+            bind:value={groupBy}
+            placeholder="None"
+          />
+        </div>
+      </div>
+    {/if}
+    <div class="buttons">
+      <ActionButton
+        quiet
+        icon="Add"
+        on:click={addCalc}
+        disabled={calculations.length >= 5}
+      >
+        Add calculation
+      </ActionButton>
+    </div>
+    <InfoDisplay
+      icon="Help"
+      quiet
+      body="Calculations only work with numeric columns and a maximum of 5 calculations can be added at once."
```

Comment:
Is this a hard technical limitation or soft limit?

Reply:
Soft limit as far as I know. This was just specified in the ticket, but I'm not sure if there's any real reason for it other than to limit the potential performance hit of running these.